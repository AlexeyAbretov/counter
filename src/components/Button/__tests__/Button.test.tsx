/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { it, describe, expect, vi } from 'vitest';

import { theme } from '@theme';
import { Button, ButtonProps } from '..';

const getUi = (props?: ButtonProps): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Button
        title="I am button"
        type="decrease"
        onClick={vi.fn()}
        {...(props || {})}
      />
    </ThemeProvider>
  );
};

describe('Button tests', () => {
  it('should render correct decrease button', () => {
    const { container } = render(getUi());

    expect(container).toMatchSnapshot();
  });

  it('should render correct increase button', () => {
    const { container } = render(
      getUi({
        type: 'increase',
      }),
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correct reset button', () => {
    const { container } = render(
      getUi({
        type: 'reset',
      }),
    );

    expect(container).toMatchSnapshot();
  });

  it('should handle onClick', () => {
    const onClick = vi.fn();

    const { getByText } = render(
      getUi({
        type: 'reset',
        onClick,
      }),
    );

    const button = getByText('I am button');

    fireEvent.click(button);

    expect(onClick).toBeCalled();
    expect(onClick).toBeCalledTimes(1);
  });

  it('should not handle onClick', () => {
    const onClick = vi.fn();

    const { getByText } = render(
      getUi({
        type: 'reset',
        onClick,
        isDisabled: true,
      }),
    );

    const button = getByText('I am button');

    fireEvent.click(button);

    expect(onClick).not.toBeCalled();
    expect(onClick).toBeCalledTimes(0);
  });
});
