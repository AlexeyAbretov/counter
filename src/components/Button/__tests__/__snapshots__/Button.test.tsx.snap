// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button tests > should render correct decrease button 1`] = `
<div>
  <button
    class="sc-bcXHqe hXcyCH"
  >
    I am button
  </button>
</div>
`;

exports[`Button tests > should render correct increase button 1`] = `
<div>
  <button
    class="sc-bcXHqe hVFLLR"
  >
    I am button
  </button>
</div>
`;

exports[`Button tests > should render correct reset button 1`] = `
<div>
  <button
    class="sc-bcXHqe gInzaI"
  >
    I am button
  </button>
</div>
`;
